File: src/main/java/net/swamphut/swampium/core/swobject/dependency/ServiceProvider.java
package net.swamphut.swampium.core.swobject.dependency;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
public @interface ServiceProvider {
    Class<?>[] provide() default {};
}

File: src/main/java/net/swamphut/swampium/core/swobject/dependency/ServiceProvider.kt
package net.swamphut.swampium.core.swobject.dependency

import kotlin.reflect.KClass

@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE)
annotation class ServiceProvider(val provide: Array<KClass<*>> = [])
